Augur v25 blog post [#1003] (#1004)
nextstrain/nextstrain.org: static-site/content/blog/2024-08-30-augur-v25-features.md

the past few months. We wanted to highlight some of the significant
feature improvements in these new releases.

### Excel and OpenOffice support

One of the features we are most excited to announce is that the
[`augur curate`][] command now supports both Excel (`.xlsx` and
`.xls`) and OpenOffice (`.ods`) files as metadata inputs. This allows
the easy conversion of Excel or OpenOffice files into the expected
metadata TSV used by other Augur commands. *This new feature is
available in Augur v25.2 and later.*

### Merging multiple metadata files

We are also very pleased to announce a new command, [`augur merge`][],
which allows for generalized merging of two or more metadata tables
based on a common field. We hope that this will make curation of
metadata inputs significantly easier. We also expect that this will
facilitate starting with existing pathogen repos with curated data
available via [data.nextstrain.org][] and “spiking in” extra,
non-public metadata for additional analysis. We plan to extend this
functionality to support merging of sequence files in a future Augur
release. *The merge feature is available in Augur v25.3 and later.*

> ### Weighted filtering

Reviewer: ```suggestion
### Weighted sampling
```

> ### Weighted filtering
> 
> We have extended `augur filter` with a new flag,
> [`--group-by-weights`][], that adds weighted subsampling. This allows

Reviewer: Sorry I didn't catch this when it was added in the initial draft. I've called this "weighted sampling" in the [guide](https://docs.nextstrain.org/en/latest/guides/bioinformatics/filtering-and-subsampling.html) and it would be good to keep consistent.

```suggestion
### Weighted sampling

We have extended `augur filter` with a new flag,
[`--group-by-weights`][], that adds weighted sampling. This allows
```

subsampling to follow quantities like population size or case counts
to facilitate more representative analyses. Or conversely, this allows
for sampling of a specific “focal” region with more intensity than
other “contextual” regions. This is described in detail in an updated
[Filtering and Subsampling guide][], as well as the help text of the

> \`--group-by-weights\` flag. *This new flag is available in Augur
> v25.3 and later.*

Reviewer: Same as https://github.com/nextstrain/nextstrain.org/pull/1004#discussion_r1735086676:

```suggestion
`--group-by-weights` flag. *This new flag is available in Augur
v25.3 and later.*
```

programs can do I/O “the Augur way”, including transparent handling of
compression formats and newlines consistent with the rest of Augur. We
hope that exposing this functionality will make it easier to integrate
external programs into Augur pipelines in a consistent, convenient
way. *These new commands are available in Augur v25.3 and later.*

### Six new augur curate commands for transforming metadata

We added a number of new sub commands to [`augur curate`][]. These
commands ease the manipulation of various sorts of dataset metadata:

* [`augur curate abbreviate-authors`][] abbreviates author lists to
  "&lt;first author&gt;, et al."
* [`augur curate apply-geolocation-rules`][] applies user-curated
  geolocation rules to geolocation fields
* [`augur curate apply-record-annotations`][] applies user-curated
  annotations to existing fields
* [`augur curate parse-genbank-location`][] parses the `geo_loc_name`
  field from GenBank records
* [`augur curate rename`][] renames field / column names.
* [`augur curate transform-strain-name`][] filters strain names using
  a regular expression

*All of these new sub-commands are available in Augur v25.0 and later.*

> ### Limited numerical precision to reduce file size

Reviewer 2: Suggestion for title:

### Reduction in Auspice JSON sizes of around 30%

Suggestion of added text:

"This will improve page load times in Auspice and nextstrain.org."
